router/network_router.go

```diff
@@ -168,6 +153,27 @@ func (router *NetworkRouter) handleForwardedPacket(key ForwardPacketKey) FlowOp
 	}
 }
 
+func (router *NetworkRouter) learnMAC(key PacketKey, srcPeer *mesh.Peer, isLocalMAC bool) {
+	srcMac := net.HardwareAddr(key.SrcMAC[:])
+
+	switch newSrcMac, conflictPeer := router.Macs.Add(srcMac, srcPeer); {
+	case newSrcMac && isLocalMAC:
+		log.Print("Discovered local MAC ", srcMac)
+	case newSrcMac:
+		log.Print("Discovered remote MAC ", srcMac, " at ", srcPeer)
+	case conflictPeer != nil:
+		// If it's a local MAC, then the MAC cache has an entry for the source
+		// MAC associated with another peer. This can happen when a MAC has moved.
+		if isLocalMAC {
+			log.Print("Discovered local MAC ", srcMac, " (was at ", conflictPeer, ")")
+		} else {
+			log.Print("Discovered remote MAC ", srcMac, " at ", srcPeer, " (was at ", conflictPeer, ")")
+		}
+		// We need to clear out any flows with the MAC as the source.
+		router.Overlay.(NetworkOverlay).InvalidateRoutes()
+	}
+}
+
```

Comment:
I wouldn't bother with the isLocal distinction. Just change the message wording to be more generic.